enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -147,7 +147,8 @@ class AdjointGenerator
   }
 
   llvm::Value *MPI_TYPE_SIZE(llvm::Value *DT, llvm::IRBuilder<> &B,
-                             llvm::Type *intType) {
+                             llvm::Type *intType,
+                             llvm::BasicBlock *allocBlock = nullptr) {
```

Comment:
When handling MPI_Allreduce using min/max op, I replace the call in the primal code with an MPI_Allreduce using the minloc/maxloc (+ some loops for copying data, and mallocs for temporary arrays). I'm not sure if it is the best strategy, but construct this replacement code inside a function, similar to `getOInsertCheckedFree`. Because I'm constructing the replacement code in a function I added the extra parameter `allocaBlock` to MPI_TYPE_SIZE etc so that I could use them in this case.